src/peep/peep.c

```diff
@@ -7131,70 +7131,115 @@ uint8 sub_69A60A(rct_peep* peep){
 
 /**
  *
+ *  rct2: 0x0069A997
+ */
+static int sub_69A997(sint16 x, sint16 y, uint8 z, int direction, rct_map_element *dest_map_element, int test_edge) {
+	int eax = x, ebx = direction, ecx = y, edx = z, esi = (int) dest_map_element, edi = 0xFFFF, ebp = test_edge;
+	RCT2_CALLFUNC_X(0x0069A997, &eax, &ebx, &ecx, &edx, &esi, &edi, &ebp);
```

Comment:
`dh` appears to be set to 0 by the calling function `pathfind_choose_direction` but then sub_69A997 calls itself recursively with different values of `dh`.